scio-avro/src/main/scala/com/spotify/scio/avro/AvroIO.scala

```diff
@@ -220,6 +220,45 @@ final case class GenericRecordIO[T: ClassTag: Coder](path: String, schema: Schem
     GenericRecordTap[T](ScioUtil.addPartSuffix(path), schema)
 }
 
+/**
+ * Given a parseFn, read [[org.apache.avro.generic.GenericRecord GenericRecord]]
+ * and apply a function mapping [[GenericRecord => T]] before producing output.
+ * This IO applies the function at the time to de-serialising Avro GenericRecords.
+ *
+ * This IO doesn't define write, and should not be used to write Avro GenericRecords.
+ */
+final case class GenericRecordParseIO[T](path: String, parseFn: GenericRecord => T)(
+  implicit coder: Coder[T]
+) extends AvroIO[T] {
+  override type ReadP = Unit
+  override type WriteP = Nothing // Output is not defined for Avro Generic Parse IO.
+
+  override def testId: String = s"AvroIO($path)"
+
+  /**
+   * Get an SCollection[T] by applying the [[parseFn]] on
+   * [[org.apache.avro.generic.GenericRecord GenericRecord]]
+   * from an Avro file.
+   */
+  override protected def read(sc: ScioContext, params: Unit): SCollection[T] = {
+    val t = beam.AvroIO
+      .parseGenericRecords(
+        Functions
+          .serializableFn(parseFn)
+      )
+      .from(path)
+      .withCoder(CoderMaterializer.beamWithDefault(coder))
```

Comment:
Please pass the correct `ScioContext` (`sc`) here and not the default one.

Reply:
Thanks for the review. I've fixed this.